### PR TITLE
Support a configurable delay for task shuffles.

### PR DESCRIPTION
**Pull request by baconmania**

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java [hunk 1]
```diff
@@ -280,6 +280,14 @@ public void runActionOnPoll() {
     }
   }
 
+  private boolean isEligibleForShuffle(SingularityTaskId task) {
+    return (
+        isLongRunning(task)
+            && !configuration.getDoNotShuffleRequests().contains(task.getRequestId())
+            && (TimeUnit.MILLISECONDS.toMinutes(System.currentTimeMillis() - task.getStartedAt()) > configuration.getMinutesBeforeNewTaskEligibleForShuffle())
```
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java]: I wonder if we should go by the task running timestamp instead? Take out any delay that may have been caused by mesos/singularity that way

File: SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java [hunk 2]
```diff
@@ -15,6 +15,8 @@
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.stream.Collectors;
 
+import javax.ws.rs.HEAD;
```
Comment (ssalinas) [inline SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java]: leftover from merge conflict? (intellij auto import stuff?)

Comment (baconmania) [reply SingularityService/src/main/java/com/hubspot/singularity/scheduler/SingularityUsagePoller.java]: Good catch.

**Comment by ssalinas** [comment]
🚢

**Pull request merged**